accepted/0000-dependencies-change-script.md

```diff
@@ -0,0 +1,51 @@
+# Dependencies Change Script
+
+## Summary
+
+Provide the ability to automatically run a script anytime the the dependencies in a repo
+change.
+
+## Motivation
+
+It would be valuable to run validation scripts after dependencies have changed,
+and would enable automatically running compliance tooling in projects when
+dependencies are installed.
+
+For example, I work at company Rad Developers (not really 😉) and we have an
+internal auditing script `MakeSureWeDontGetSued.sh` we need to run on any
+dependencies we use in our applications. I want to be able to setup a
+package.json script that will be called automatically anytime the dependencies
+change so that compliance is an automated part of every team members' workflow.
+
+## Detailed Explanation
+
+I think one way this could be backwards compatible is by adding a `postChange`
```

Comment:
I think this should be a lifecycle called `change`, where `prechange` runs when we've determined changes will be needed, but no tree changes have been applied, and then `change` and `postchange` would be run after the tree itself gets updated (these should both run after `postinstall`).

Reply:
I would certainly like to be able to have a `prechange` and `postchange`!

What would be the difference be between a `change` and a `postchange`, though?

Reply:
Likely none, that both exist is more of an artifact of how our lifecycle scripts work. (This is the case with most lifecycle scripts.) In a perfect world, I would only call the non pre/post named lifecycle if it was _replacing_ the action taken at that point, but given the history, keeping things consistent seems better even if it's a little odd to have two lifecycles that do the same thing.